tnc/data_handler.py

```diff
@@ -261,92 +272,100 @@ def process_data(self, bytes_out, freedv, bytes_per_frame: int) -> None:
 
             # FRAME NACK
-            elif frametype == 63:
+            elif frametype == FR_TYPE.FR_NACK.value:
                 self.log.debug("[TNC] FRAME NACK RECEIVED....")
                 self.frame_nack_received(bytes_out[:-2])
 
             # BURST NACK
-            elif frametype == 64:
+            elif frametype == FR_TYPE.BURST_NACK.value:
                 self.log.debug("[TNC] BURST NACK RECEIVED....")
                 self.burst_nack_received(bytes_out[:-2])
 
             # CQ FRAME
-            elif frametype == 200:
+            elif frametype == FR_TYPE.CQ.value:
                 self.log.debug("[TNC] CQ RECEIVED....")
                 self.received_cq(bytes_out[:-2])
 
             # QRV FRAME
-            elif frametype == 201:
+            elif frametype == FR_TYPE.QRV.value:
                 self.log.debug("[TNC] QRV RECEIVED....")
                 self.received_qrv(bytes_out[:-2])
 
             # PING FRAME
-            elif frametype == 210:
+            elif frametype == FR_TYPE.PING.value:
                 self.log.debug("[TNC] PING RECEIVED....")
                 self.received_ping(bytes_out[:-2])
 
             # PING ACK
-            elif frametype == 211:
+            elif frametype == FR_TYPE.PING_ACK.value:
                 self.log.debug("[TNC] PING ACK RECEIVED....")
                 self.received_ping_ack(bytes_out[:-2])
 
             # SESSION OPENER
-            elif frametype == 221:
+            elif frametype == FR_TYPE.ARQ_SESSION_OPEN.value:
                 self.log.debug("[TNC] OPEN SESSION RECEIVED....")
                 self.received_session_opener(bytes_out[:-2])
 
             # SESSION HEARTBEAT
-            elif frametype == 222:
+            elif frametype == FR_TYPE.ARQ_SESSION_HB.value:
                 self.log.debug("[TNC] SESSION HEARTBEAT RECEIVED....")
                 self.received_session_heartbeat(bytes_out[:-2])
 
             # SESSION CLOSE
-            elif frametype == 223:
+            elif frametype == FR_TYPE.ARQ_SESSION_CLOSE.value:
                 self.log.debug("[TNC] CLOSE ARQ SESSION RECEIVED....")
                 self.received_session_close(bytes_out[:-2])
 
             # ARQ FILE TRANSFER RECEIVED!
-            elif frametype in [225, 227]:
+            elif frametype in [
+                FR_TYPE.ARQ_DC_OPEN_W.value,
+                FR_TYPE.ARQ_DC_OPEN_N.value,
+            ]:
                 self.log.debug("[TNC] ARQ arq_received_data_channel_opener")
                 self.arq_received_data_channel_opener(bytes_out[:-2])
 
             # ARQ CHANNEL IS OPENED
-            elif frametype in [226, 228]:
+            elif frametype in [
+                FR_TYPE.ARQ_DC_OPEN_ACK_W.value,
+                FR_TYPE.ARQ_DC_OPEN_ACK_N.value,
+            ]:
                 self.log.debug("[TNC] ARQ arq_received_channel_is_open")
                 self.arq_received_channel_is_open(bytes_out[:-2])
 
             # ARQ MANUAL MODE TRANSMISSION
```

Comment:
In my opinion we can remove the manual mode selection, as a automatic selection fits best, specially under really bad HF conditions. This feature works well in development environments but not under real circumstances